nautobot/extras/tests/test_views.py

```diff
@@ -2846,6 +2847,59 @@ def test_job_object_change_log_view(self):
         response = self.client.get(instance.get_changelog_url())
         self.assertBodyContains(response, f"{instance.name} - Change Log")
 
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_job_bulk_edit_default_queue_stays_default_after_one_field_update(self):
+        self.add_permissions("extras.change_job")
+        job_class_to_test = "TestPass"
+        job_description = "default job queue test"
+        job_to_test = self.model.objects.get(job_class_name=job_class_to_test)
+        queryset = self.model.objects.filter(installed=True, hidden=False, job_class_name=job_class_to_test)
+        default_job_queue = job_to_test.default_job_queue
+        pk_list = list(queryset.values_list("pk", flat=True))
+
+        data = {
+            "description": job_description,
+            "pk": pk_list,
+            "_apply": True,
+        }
+
+        self.assertHttpStatus(self.client.post(self._get_url("bulk_edit"), data), HTTPStatus.FOUND)
+        instance = self.model.objects.get(job_class_name=job_class_to_test)
+        self.assertEqual(instance.description, job_description)
+        self.assertEqual(instance.default_job_queue, default_job_queue)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_job_bulk_edit_preserve_job_queues_after_one_field_update(self):
+        self.add_permissions("extras.change_job")
+        job_class_to_test = "TestPass"
+        job_description = "job queues test"
+        job_to_test = self.model.objects.get(job_class_name=job_class_to_test)
+
+        # Simulate 3 job queues set
+        job_queues = JobQueue.objects.all()[:3]
+        job_to_test.job_queues.set(job_queues)
+        job_to_test.job_queues_override = True
+        job_to_test.save()
+
+        default_job_queue = job_to_test.default_job_queue
+        expected_job_queues = [*job_queues.values_list("pk", flat=True), default_job_queue.pk]
```

Comment:
This feels weird to me - the `default_job_queue` shouldn't be possible to set to a value that isn't in `job_queues` as a selected option. I know we had some issue around data validation (challenges of many-to-many) in this area but at the very least this feels wrong as test data.

Reply:
Somehow it's not possible: we have a signal which will add a `default_job_queue` to the `job_queues`.